src/elements/common/item/ItemOptions.tsx

```diff
@@ -66,63 +69,84 @@ const ItemOptions = ({
-                    </DropdownMenu.Trigger>
-                    <DropdownMenu.Content align="end">
-                        {isPreviewEnabled && (
-                            <DropdownMenu.Item onClick={() => onItemPreview(item)}>
-                                {formatMessage(messages.preview)}
-                            </DropdownMenu.Item>
-                        )}
-                        {isOpenEnabled && (
-                            <DropdownMenu.Item onClick={() => onItemPreview(item)}>
-                                {formatMessage(messages.open)}
-                            </DropdownMenu.Item>
-                        )}
-                        {isDeleteEnabled && (
-                            <DropdownMenu.Item onClick={() => onItemDelete(item)}>
-                                {formatMessage(messages.delete)}
-                            </DropdownMenu.Item>
-                        )}
-                        {isDownloadEnabled && (
-                            <DropdownMenu.Item onClick={() => onItemDownload(item)}>
-                                {formatMessage(messages.download)}
-                            </DropdownMenu.Item>
-                        )}
-                        {isRenameEnabled && (
-                            <DropdownMenu.Item onClick={() => onItemRename(item)}>
-                                {formatMessage(messages.rename)}
-                            </DropdownMenu.Item>
-                        )}
-                        {isShareEnabled && (
-                            <DropdownMenu.Item onClick={() => onItemShare(item)}>
-                                {formatMessage(messages.share)}
-                            </DropdownMenu.Item>
-                        )}
-                        {hasActions && hasOptions && <DropdownMenu.Separator />}
-                        {itemActions.map(({ label: actionLabel, onAction, type: actionType }) => {
-                            if (actionType && actionType !== itemType) {
-                                return null;
-                            }
-                            return (
-                                <DropdownMenu.Item key={actionLabel + actionType} onClick={() => onAction(item)}>
-                                    {actionLabel}
-                                </DropdownMenu.Item>
-                            );
-                        })}
-                    </DropdownMenu.Content>
-                </DropdownMenu.Root>
-            )}
-        </OptionsGroup>
+                    return (
+                        <DropdownMenu.Item key={actionLabel + actionType} onClick={() => onAction(item)}>
+                            {actionLabel}
+                        </DropdownMenu.Item>
+                    );
+                })}
+            </DropdownMenu.Content>
+        </DropdownMenu.Root>
     );
+
+    // TODO: Update to one `return` statement after ContentPicker has been migrated to Blueprint
```

Comment:
This implementation allows ItemOptions to be outside of the Blueprint list-item. But it won't be necessary anymore (and thus we can simplify it) after we migrate ContentPicker